### PR TITLE
feat: dockerizing api

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Description
This lets us create a docker image of the API in order to do so:

1. in your api .env file change the database url to one of the hosted dbs (I used the core hosted db connection string)
2. from the api directory in a terminal run `docker build -t api .  ` followed by `docker run --rm -p 3100:3100 api` or you can run `docker compose up`
3. start up the partner site like normal and log in you should be up and running like normal!

I used the following in order to get to this point:
https://dev.to/chukwutosin_/step-by-step-guide-setting-up-a-nestjs-application-with-docker-and-postgresql-5hei
and 
https://medium.com/@sujan.dumaru.official/dockerizing-nestjs-application-c4b25139fe4c

there is a known issue with connecting from the docker container to the locally running db. Despite trying a bunch of stuff I couldn't get it to work, but I also don't expect us to use docker for local development anyway so shouldn't be a big problem

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Dockerfile
+.dockerignore
+node_modules
+dist
+coverage
```
Comment (mcgarrye) [inline api/.dockerignore:5]: nit: EOF

File: api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# run build commands
+RUN yarn prisma generate
+RUN yarn build
+
+# Expose port 3101 for api
+EXPOSE 3101
+
+# Start api
+CMD ["yarn", "dev"]
```
Comment (mcgarrye) [inline api/Dockerfile:31]: nit: EOF

File: api/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# Copy .env
+COPY .env ./
+
+# run build commands
+RUN yarn prisma generate
+RUN yarn build
+
+# Expose port 3101 for api
+EXPOSE 3101
```
Comment (mcgarrye) [inline api/Dockerfile:28]: 3100 pretty please

**Review by mcgarrye — APPROVED** [review]

**Comment by YazeedLoonat** [comment]
Hey @mcgarrye updated everything and I've returned back to the 3100 port :D

**Pull request merged**